Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp

```diff
@@ -27,35 +28,39 @@ TimeZone::TimeZone(String identifier, Object& prototype)
 }
 
 // 11.1.1 IsValidTimeZoneName ( timeZone ), https://tc39.es/proposal-temporal/#sec-isvalidtimezonename
-// NOTE: This is the minimum implementation of IsValidTimeZoneName, supporting only the "UTC" time zone.
+// 15.1.1 IsValidTimeZoneName ( timeZone ), https://tc39.es/proposal-temporal/#sup-isvalidtimezonename
 bool is_valid_time_zone_name(String const& time_zone)
 {
-    // 1. Assert: Type(timeZone) is String.
-
-    // 2. Let tzText be ! StringToCodePoints(timeZone).
-    // 3. Let tzUpperText be the result of toUppercase(tzText), according to the Unicode Default Case Conversion algorithm.
-    // 4. Let tzUpper be ! CodePointsToString(tzUpperText).
-    auto tz_upper = time_zone.to_uppercase();
+    // The IsValidTimeZoneName abstract operation verifies that the timeZone argument (which must be a String value) represents a valid Zone or Link name of the IANA Time Zone Database.
+    // The abstract operation returns true if timeZone, converted to upper case as described in 6.1, is equal to one of the Zone or Link names of the IANA Time Zone Database, converted to upper case as described in 6.1. It returns false otherwise.
 
-    // 5. If tzUpper and "UTC" are the same sequence of code points, return true.
-    if (tz_upper == "UTC")
-        return true;
+    if (!::TimeZone::built_with_time_zone_data())
+        return time_zone.to_uppercase() == "UTC"sv;
 
-    // 6. Return false.
-    return false;
+    return ::TimeZone::time_zone_from_string(time_zone).has_value();
 }
 
 // 11.1.2 CanonicalizeTimeZoneName ( timeZone ), https://tc39.es/proposal-temporal/#sec-canonicalizetimezonename
-// NOTE: This is the minimum implementation of CanonicalizeTimeZoneName, supporting only the "UTC" time zone.
+// 15.1.2 CanonicalizeTimeZoneName ( timeZone ), https://tc39.es/proposal-temporal/#sup-canonicalizetimezonename
 String canonicalize_time_zone_name(String const& time_zone)
 {
     // 1. Assert: Type(timeZone) is String.
 
     // 2. Assert: ! IsValidTimeZoneName(timeZone) is true.
     VERIFY(is_valid_time_zone_name(time_zone));
 
-    // 3. Return "UTC".
-    return "UTC";
+    if (!::TimeZone::built_with_time_zone_data()) {
+        // 3. Return "UTC".
+        return "UTC"sv;
+    }
+
+    // 3. Let ianaTimeZone be the Zone or Link name of the IANA Time Zone Database such that timeZone, converted to upper case as described in 6.1, is equal to ianaTimeZone, converted to upper case as described in 6.1.
+    // 4. If ianaTimeZone is a Link name, let ianaTimeZone be the corresponding Zone name as specified in the "backward" file of the IANA Time Zone Database.
+    // 5. If ianaTimeZone is "Etc/UTC" or "Etc/GMT", return "UTC".
```

Comment:
Since this step is explicit here I assume that the TZDB doesn't actually have this alias, in which case, is this correct?

Reply:
This is already done by `canonicalize_time_zone()`. Will add a comment :+1: